docs/cookbook/extending-product-list.md

```diff
@@ -0,0 +1,127 @@
+# Extending Product List
+
+If you want add more information to product lists, you need to do a little bit more than [extending an entity](/docs/cookbook/adding-new-attribute-to-an-entity.md).
+That is because the frontend product lists leverage [the read model concept](/docs/model/introduction-to-read-model.md), i.e. special view objects are used instead of common Doctrine entities in the Twig templates.
+
+This cookbook describes how to extend the frontend product lists with step by step instructions.
+
+## Display a brand name in a product view on the product list
+
+### 1. Extend `ListedProductView` class and add a new attribute.
+
+The class encapsulates the data that are needed for displaying products on FE product lists.
+We want to display a brand name for each product so we need to add the attribute to the class.
+
+```php
+declare(strict_types=1);
+
+namespace Shopsys\ShopBundle\Model\Product\View;
+
+use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice;
+use Shopsys\ReadModelBundle\Image\ImageView;
+use Shopsys\ReadModelBundle\Product\Action\ProductActionView;
+use Shopsys\ReadModelBundle\Product\Listed\ListedProductView as BaseListedProductView;
+
+class ListedProductView extends BaseListedProductView
+{
+    /**
+     * @var string|null
+     */
+    protected $brandName;
+
+    /**
+     * @param int $id
+     * @param string $name
+     * @param string|null $brandName
+     * @param string|null $shortDescription
+     * @param string $availability
+     * @param \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice $sellingPrice
+     * @param array $flagIds
+     * @param \Shopsys\ReadModelBundle\Product\Action\ProductActionView $action
+     * @param \Shopsys\ReadModelBundle\Image\ImageView|null $image
+     */
+    public function __construct(
+        int $id,
+        string $name,
+        ?string $brandName,
+        ?string $shortDescription,
+        string $availability,
+        ProductPrice $sellingPrice,
+        array $flagIds,
+        ProductActionView $action,
+        ?ImageView $image
+    ) {
+        parent::__construct($id, $name, $shortDescription, $availability, $sellingPrice, $flagIds, $action, $image);
+
+        $this->brandName = $brandName;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getBrandName(): ?string
+    {
+        return $this->brandName;
+    }
+}
+```
+
+### 2. Extend `ListedProductViewFactory` so it returns the new required data
+
+The class is responsible for creating the view object. We need to ensure that the objects is created with proper brand name. We are able to get the brand name from the product entity, so we just need to overwrite `createFromProduct()` method.  
+
+```php
+declare(strict_types=1);
+
+namespace Shopsys\ShopBundle\Model\Product\View;
+
+use Shopsys\FrameworkBundle\Model\Product\Product;
```

Comment:
```suggestion
use Shopsys\ShopBundle\Model\Product\Product;
```

Reply:
I am not able to do that - it is not possible to overwrite the parent method definition